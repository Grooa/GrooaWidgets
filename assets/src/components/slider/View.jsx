import PropTypes from 'prop-types';

const View = ({
  image,
  title,
  body = null,
  url = null,
  urlLabel = null,
}) => <div>
  <div className="gw-slider-slide-image-container">
    <div className="gw-slider-slide-image" style={{ backgroundImage: `url("${image}")` }}/>
  </div>
  <div className="gw-slider-slide-content">
    <h3 className="gw-slider-slide-title">{title}</h3>
    {body !== null && <div className="gw-slider-slide-body">{body}</div>}

    {url !== null && <a href={url} className="gw-btn gw-btn-pill gw-slider-slide-button">
      {urlLabel || 'Go to page'}
    </a>}
  </div>
</div>;

View.propTypes = {
  image: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  body: PropTypes.string,
  url: PropTypes.string,
  urlLabel: PropTypes.string,
};

export default View;
