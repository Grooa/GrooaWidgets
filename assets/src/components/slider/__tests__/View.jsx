import { render } from 'enzyme';
import View from '../View';

describe('<View />', () => {
  let sliderView;

  beforeEach(() => {
    sliderView = {
      image: 'http://test.com/test.jpg',
      url: '/testTest',
      urlLabel: 'Go to Test',
      title: 'Hello world',
      body: 'Lorem lipsum dolor sit amed',
    };
  });

  it('should render properly', () => {
    const wrapper = render(<View {...sliderView} />);

    expect(wrapper).toMatchSnapshot();
  });
});
