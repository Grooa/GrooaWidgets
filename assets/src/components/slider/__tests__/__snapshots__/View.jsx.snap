// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<View /> should render properly 1`] = `
<div>
  <div
    class="gw-slider-slide-image-container"
  >
    <div
      class="gw-slider-slide-image"
      style="background-image:url(\\"http://test.com/test.jpg\\")"
    />
  </div>
  <div
    class="gw-slider-slide-content"
  >
    <h3
      class="gw-slider-slide-title"
    >
      Hello world
    </h3>
    <div
      class="gw-slider-slide-body"
    >
      Lorem lipsum dolor sit amed
    </div>
    <a
      class="gw-btn gw-btn-pill gw-slider-slide-button"
      href="/testTest"
    >
      Go to Test
    </a>
  </div>
</div>
`;
